src/taco-dependency-installer/test/androidSdkInstaller.ts

```diff
@@ -121,12 +158,193 @@ describe("AndroidSdkInstaller telemetry", () => {
                 }
             ];
 
-            return androidSdkInstaller.run()
-                .then(() => Q.reject(new Error("Should have gotten a rejection in this test")), () => {
-                    return fakeTelemetryHelper.getAllSentEvents().then((allSentEvents: TelemetryEvent[]) => {
-                        allSentEvents.should.eql(expectedTelemetry);
-                    });
-                }).done(done, done);
+            return telemetryGeneratedShouldBe(expectedTelemetry, /Error while executing/, done);
+        });
+    });
+
+    describe("installation", () => {
+        it("generates telemetry error if there is no install location", (done: MochaDone) => {
+            fakeProcess.fakeMacOS();
+            steps.install = true; // We only test this step on this test
+            installTo = ""; // We don't have an install location
+
+            var expectedTelemetry: TacoUtility.ICommandTelemetryProperties[] = [
+                {
+                    "initialStep.time": { isPii: false, value: "2000" },
+                    step: { isPii: false, value: "initialStep" }
+                },
+                {
+                    "error.description": { isPii: false, value: "NeedInstallDestination on installDefault" },
+                    "install.time": { isPii: false, value: "2000" },
+                    lastStepExecuted: { isPii: false, value: "install" },
+                    step: { isPii: false, value: "install" },
+                    time: { isPii: false, value: "3000" }
+                }];
+
+            return telemetryGeneratedShouldBe(expectedTelemetry, /An installation destination is needed for this dependency/, done);
```

Comment:
We should set the resource manager to test mode in the before() handler so that we only have to deal with string IDs, not full strings